src/checkers/inference/InferenceAnnotatedTypeFactory.java

```diff
@@ -576,5 +580,33 @@ protected InferenceViewpointAdapter createViewpointAdapter() {
     protected Set<? extends AnnotationMirror> getDefaultTypeDeclarationBounds() {
         return realTypeFactory.getQualifierHierarchy().getTopAnnotations();
     }
+
+    /**
+     * Get the annotation from the class declaration.
+     * @param type a type
+     * @return the {@link VarAnnot} on the class bound, or the result from
+     * {@link org.checkerframework.framework.type.AnnotatedTypeFactory#getTypeDeclarationBounds}
+     * if the class is not handled by the {@link VariableAnnotator}.
+     */
+    @Override
+    public Set<AnnotationMirror> getTypeDeclarationBounds(TypeMirror type) {
```

Comment:
Add comment to the super type method to clarify the behavior of this method, 
and refer to @UpperBoundFor for clarification